### ISSUE TITLE
Could not poll the database: could not find driver on first load of /public/

### ISSUE THREAD

**Issue by Hauptstadt** [firefly-iii/firefly-iii#4202]
I am running Firefly III version 5.4.6

**Description**
Hi there, I'm back.  As a refresher: 
I am trying to install Firefly 5.4.6 on my Synology NAS in the native MariaDB / Web Station apps
I am *not* using Docker
  
When I try to run firefly-iii/public for the first time, I get the following error on my screen:

Could not poll the database: could not find driver (SQL: select \`id\`, \`name\`, \`data\` from \`configuration\` where \`name\` = is_demo_site and \`configuration\`.\`deleted_at\` is null limit 1)  
  
When I run the query (is_demo_site = false) in my phpmyadmin, the query returns one line from the configuration table, so I know it exists and functions, so the mess up is somewhere in between the mariadb server and the request, I'm assuming.   
  
Any ideas? 

**Extra info**
 

**Bonus points**

- I have read the FAQ at https://bit.ly/FF3-FAQ
👍
- I added a screenshot
![image](https://user-images.githubusercontent.com/64873401/103383436-ff422c80-4af2-11eb-9dee-f8134f40e4bf.png)

- I added log files 

```[2020-12-30 23:03:45] local.INFO: Could not create or generate installation ID. Do not continue.  
[2020-12-30 23:03:45] local.ERROR: Exception is: {"class":"FireflyIII\\Exceptions\\FireflyException","errorMessage":"Could not poll the database: could not find driver (SQL: select `id`, `name`, `data` from `configuration` where `name` = is_demo_site and `configuration`.`deleted_at` is null limit 1)","time":"Wed, 30 Dec 2020 23:03:45 +0100","file":"\/volume1\/web\/firefly-iii\/app\/Support\/FireflyConfig.php","line":90,"code":0,"version":"5.4.6","url":"http:\/\/192.168.2.106:8080\/firefly-iii\/public","userAgent":"Mozilla\/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/87.0.4280.88 Safari\/537.36","json":true}  
[2020-12-30 23:03:45] local.DEBUG: Message-ID: <746ecbfe731b781c42502749ab5cfc5e@192.168.2.106>
Date: Wed, 30 Dec 2020 23:03:45 +0100
From: Firefly III Mailer <changeme@example.com>
MIME-Version: 1.0
Content-Type: multipart/alternative;
 boundary="_=_swift_1609365825_bca2e58c3ab905391bf6ed63ab183242_=_"

<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
    <meta content="text/html; charset=UTF-8" http-equiv="Content-Type">

</head>
<body>
<p style="font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;font-size:13px;">
    Hi there,
</p>
<p style="font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;font-size:13px;">
    Firefly III v5.4.6 ran into an error: <span style="font-family: monospace;">Could not poll the database: could not find driver (SQL: select `id`, `name`, `data` from `configuration` where `name` = is_demo_site and `configuration`.`deleted_at` is null limit 1)</span>.
</p>

<p style="font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;font-size:13px;">
    The error was of type &quot;FireflyIII\Exceptions\FireflyException&quot;.
</p>

<p style="font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;font-size:13px;">
    The error occurred on/at: Wed, 30 Dec 2020 23:03:45 +0100.

</p>

<p style="font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;font-size:13px;">
    This error occurred in file "<span style="font-family: monospace;">/volume1/web/firefly-iii/app/Support/FireflyConfig.php</span>" on line 90 with code 0.
</p>

<p style="font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;font-size:13px;">
            There was no user logged in for this error or no user was detected.
    </p>

<p style="font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;font-size:13px;">
    The IP address related to this error is: <br />
    URL is: http://192.168.2.106:8080/firefly-iii/public<br />
    User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36
</p>

<p style="font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;font-size:13px;">
    The full stacktrace is below. If you think this is a bug in Firefly III, you can forward this message to <a href="mailto:james@firefly-iii.org?subject=BUG!">james@firefly-iii.org</a>. This can help fix the bug you just encountered.
</p>
<p style="font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;font-size:13px;">
    If you prefer, you can also open a new issue on <a href="https://github.com/firefly-iii/firefly-iii/issues">GitHub</a>.
</p>

<p style="font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;font-size:13px;">
    The full stacktrace is below:</p>
<p style="font-family: monospace;font-size:11px;color:#aaa">
    #0 /volume1/web/firefly-iii/app/Http/Controllers/Controller.php(58): FireflyIII\Support\FireflyConfig-&gt;get(&#039;is_demo_site&#039;, false)<br />
#1 /volume1/web/firefly-iii/app/Http/Controllers/HomeController.php(55): FireflyIII\Http\Controllers\Controller-&gt;__construct()<br />
#2 [internal function]: FireflyIII\Http\Controllers\HomeController-&gt;__construct()<br />
#3 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php(849): ReflectionClass-&gt;newInstanceArgs(Array)<br />
#4 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php(691): Illuminate\Container\Container-&gt;build(&#039;FireflyIII\\Http...&#039;)<br />
#5 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(796): Illuminate\Container\Container-&gt;resolve(&#039;FireflyIII\\Http...&#039;, Array, true)<br />
#6 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php(637): Illuminate\Foundation\Application-&gt;resolve(&#039;FireflyIII\\Http...&#039;, Array)<br />
#7 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(781): Illuminate\Container\Container-&gt;make(&#039;FireflyIII\\Http...&#039;, Array)<br />
#8 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Route.php(269): Illuminate\Foundation\Application-&gt;make(&#039;FireflyIII\\Http...&#039;)<br />
#9 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Route.php(990): Illuminate\Routing\Route-&gt;getController()<br />
#10 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Route.php(951): Illuminate\Routing\Route-&gt;controllerMiddleware()<br />
#11 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(708): Illuminate\Routing\Route-&gt;gatherMiddleware()<br />
#12 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(684): Illuminate\Routing\Router-&gt;gatherRouteMiddleware(Object(Illuminate\Routing\Route))<br />
#13 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(668): Illuminate\Routing\Router-&gt;runRouteWithinStack(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request))<br />
#14 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(634): Illuminate\Routing\Router-&gt;runRoute(Object(Illuminate\Http\Request), Object(Illuminate\Routing\Route))<br />
#15 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(623): Illuminate\Routing\Router-&gt;dispatchToRoute(Object(Illuminate\Http\Request))<br />
#16 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(166): Illuminate\Routing\Router-&gt;dispatch(Object(Illuminate\Http\Request))<br />
#17 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\Foundation\Http\Kernel-&gt;Illuminate\Foundation\Http\{closure}(Object(Illuminate\Http\Request))<br />
#18 /volume1/web/firefly-iii/app/Http/Middleware/InstallationId.php(53): Illuminate\Pipeline\Pipeline-&gt;Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))<br />
#19 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): FireflyIII\Http\Middleware\InstallationId-&gt;handle(Object(Illuminate\Http\Request), Object(Closure))<br />
#20 /volume1/web/firefly-iii/vendor/fideloper/proxy/src/TrustProxies.php(57): Illuminate\Pipeline\Pipeline-&gt;Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))<br />
#21 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Fideloper\Proxy\TrustProxies-&gt;handle(Object(Illuminate\Http\Request), Object(Closure))<br />
#22 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\Pipeline\Pipeline-&gt;Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))<br />
#23 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\Foundation\Http\Middleware\TransformsRequest-&gt;handle(Object(Illuminate\Http\Request), Object(Closure))<br />
#24 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\Pipeline\Pipeline-&gt;Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))<br />
#25 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\Foundation\Http\Middleware\TransformsRequest-&gt;handle(Object(Illuminate\Http\Request), Object(Closure))<br />
#26 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\Pipeline\Pipeline-&gt;Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))<br />
#27 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\Foundation\Http\Middleware\ValidatePostSize-&gt;handle(Object(Illuminate\Http\Request), Object(Closure))<br />
#28 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(87): Illuminate\Pipeline\Pipeline-&gt;Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))<br />
#29 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\Foundation\Http\Middleware\PreventRequestsDuringMaintenance-&gt;handle(Object(Illuminate\Http\Request), Object(Closure))<br />
#30 /volume1/web/firefly-iii/app/Http/Middleware/SecureHeaders.php(51): Illuminate\Pipeline\Pipeline-&gt;Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))<br />
#31 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): FireflyIII\Http\Middleware\SecureHeaders-&gt;handle(Object(Illuminate\Http\Request), Object(Closure))<br />
#32 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\Pipeline\Pipeline-&gt;Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))<br />
#33 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(141): Illuminate\Pipeline\Pipeline-&gt;then(Object(Closure))<br />
#34 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(110): Illuminate\Foundation\Http\Kernel-&gt;sendRequestThroughRouter(Object(Illuminate\Http\Request))<br />
#35 /volume1/web/firefly-iii/public/index.php(76): Illuminate\Foundation\Http\Kernel-&gt;handle(Object(Illuminate\Http\Request))<br />
#36 {main}
</p>
<p style="font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;font-size:13px;">
    Beep boop,
</p>
<p style="font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;font-size:13px;">
    The Firefly III Mail Robot
</p>

<p style="font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;font-size:11px;color:#aaa;">
    PS: This message was sent because a request from IP 192.168.2.109 triggered it.
</p>

</body>
</html>


Content-Type: text/plain; charset=utf-8
Content-Transfer-Encoding: quoted-printable

Hi there,

Firefly III v5.4.6 ran into an error: Could not poll the database: could not find driver (SQL: select `id`, `name`, `data` from `configuration` where `name` = is_demo_site and `configuration`.`deleted_at` is null limit 1).

The error was of type "FireflyIII\Exceptions\FireflyException".

The error occurred on/at: Wed, 30 Dec 2020 23:03:45 +0100.

This error occurred in file "/volume1/web/firefly-iii/app/Support/FireflyConfig.php" on line 90 with code 0.

There was no user logged in for this error or no user was detected.

The IP address related to this error is: 
URL is: http://192.168.2.106:8080/firefly-iii/public
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.88 Safari/537.36

The full stacktrace is below. If you think this is a bug in Firefly III, you can forward this message to james@firefly-iii.org. This can help fix the bug you just encountered.

If you prefer, you can also open a new issue on https://github.com/firefly-iii/firefly-iii/issues.

The full stacktrace is below:

#0 /volume1/web/firefly-iii/app/Http/Controllers/Controller.php(58): FireflyIII\Support\FireflyConfig->get('is_demo_site', false)
#1 /volume1/web/firefly-iii/app/Http/Controllers/HomeController.php(55): FireflyIII\Http\Controllers\Controller->__construct()
#2 [internal function]: FireflyIII\Http\Controllers\HomeController->__construct()
#3 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php(849): ReflectionClass->newInstanceArgs(Array)
#4 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php(691): Illuminate\Container\Container->build('FireflyIII\\Http...')
#5 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(796): Illuminate\Container\Container->resolve('FireflyIII\\Http...', Array, true)
#6 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php(637): Illuminate\Foundation\Application->resolve('FireflyIII\\Http...', Array)
#7 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(781): Illuminate\Container\Container->make('FireflyIII\\Http...', Array)
#8 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Route.php(269): Illuminate\Foundation\Application->make('FireflyIII\\Http...')
#9 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Route.php(990): Illuminate\Routing\Route->getController()
#10 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Route.php(951): Illuminate\Routing\Route->controllerMiddleware()
#11 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(708): Illuminate\Routing\Route->gatherMiddleware()
#12 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(684): Illuminate\Routing\Router->gatherRouteMiddleware(Object(Illuminate\Routing\Route))
#13 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(668): Illuminate\Routing\Router->runRouteWithinStack(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request))
#14 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(634): Illuminate\Routing\Router->runRoute(Object(Illuminate\Http\Request), Object(Illuminate\Routing\Route))
#15 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(623): Illuminate\Routing\Router->dispatchToRoute(Object(Illuminate\Http\Request))
#16 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(166): Illuminate\Routing\Router->dispatch(Object(Illuminate\Http\Request))
#17 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}(Object(Illuminate\Http\Request))
#18 /volume1/web/firefly-iii/app/Http/Middleware/InstallationId.php(53): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#19 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): FireflyIII\Http\Middleware\InstallationId->handle(Object(Illuminate\Http\Request), Object(Closure))
#20 /volume1/web/firefly-iii/vendor/fideloper/proxy/src/TrustProxies.php(57): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#21 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Fideloper\Proxy\TrustProxies->handle(Object(Illuminate\Http\Request), Object(Closure))
#22 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#23 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\Foundation\Http\Middleware\TransformsRequest->handle(Object(Illuminate\Http\Request), Object(Closure))
#24 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#25 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\Foundation\Http\Middleware\TransformsRequest->handle(Object(Illuminate\Http\Request), Object(Closure))
#26 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#27 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\Foundation\Http\Middleware\ValidatePostSize->handle(Object(Illuminate\Http\Request), Object(Closure))
#28 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(87): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#29 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\Foundation\Http\Middleware\PreventRequestsDuringMaintenance->handle(Object(Illuminate\Http\Request), Object(Closure))
#30 /volume1/web/firefly-iii/app/Http/Middleware/SecureHeaders.php(51): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#31 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): FireflyIII\Http\Middleware\SecureHeaders->handle(Object(Illuminate\Http\Request), Object(Closure))
#32 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#33 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(141): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#34 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(110): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter(Object(Illuminate\Http\Request))
#35 /volume1/web/firefly-iii/public/index.php(76): Illuminate\Foundation\Http\Kernel->handle(Object(Illuminate\Http\Request))
#36 {main}

Beep boop,

The Firefly III Mail Robot

PS: This message was sent because a request from IP 192.168.2.109 triggered it.
  
[2020-12-30 23:03:45] local.ERROR: Could not poll the database: could not find driver (SQL: select `id`, `name`, `data` from `configuration` where `name` = is_demo_site and `configuration`.`deleted_at` is null limit 1) {"exception":"[object] (FireflyIII\\Exceptions\\FireflyException(code: 0): Could not poll the database: could not find driver (SQL: select `id`, `name`, `data` from `configuration` where `name` = is_demo_site and `configuration`.`deleted_at` is null limit 1) at /volume1/web/firefly-iii/app/Support/FireflyConfig.php:90)
[stacktrace]
#0 /volume1/web/firefly-iii/app/Http/Controllers/Controller.php(58): FireflyIII\\Support\\FireflyConfig->get('is_demo_site', false)
#1 /volume1/web/firefly-iii/app/Http/Controllers/HomeController.php(55): FireflyIII\\Http\\Controllers\\Controller->__construct()
#2 [internal function]: FireflyIII\\Http\\Controllers\\HomeController->__construct()
#3 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php(849): ReflectionClass->newInstanceArgs(Array)
#4 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php(691): Illuminate\\Container\\Container->build('FireflyIII\\\\Http...')
#5 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(796): Illuminate\\Container\\Container->resolve('FireflyIII\\\\Http...', Array, true)
#6 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php(637): Illuminate\\Foundation\\Application->resolve('FireflyIII\\\\Http...', Array)
#7 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(781): Illuminate\\Container\\Container->make('FireflyIII\\\\Http...', Array)
#8 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Route.php(269): Illuminate\\Foundation\\Application->make('FireflyIII\\\\Http...')
#9 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Route.php(990): Illuminate\\Routing\\Route->getController()
#10 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Route.php(951): Illuminate\\Routing\\Route->controllerMiddleware()
#11 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(708): Illuminate\\Routing\\Route->gatherMiddleware()
#12 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(684): Illuminate\\Routing\\Router->gatherRouteMiddleware(Object(Illuminate\\Routing\\Route))
#13 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(668): Illuminate\\Routing\\Router->runRouteWithinStack(Object(Illuminate\\Routing\\Route), Object(Illuminate\\Http\\Request))
#14 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(634): Illuminate\\Routing\\Router->runRoute(Object(Illuminate\\Http\\Request), Object(Illuminate\\Routing\\Route))
#15 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(623): Illuminate\\Routing\\Router->dispatchToRoute(Object(Illuminate\\Http\\Request))
#16 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(166): Illuminate\\Routing\\Router->dispatch(Object(Illuminate\\Http\\Request))
#17 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\\Foundation\\Http\\Kernel->Illuminate\\Foundation\\Http\\{closure}(Object(Illuminate\\Http\\Request))
#18 /volume1/web/firefly-iii/app/Http/Middleware/InstallationId.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#19 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): FireflyIII\\Http\\Middleware\\InstallationId->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#20 /volume1/web/firefly-iii/vendor/fideloper/proxy/src/TrustProxies.php(57): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#21 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Fideloper\\Proxy\\TrustProxies->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#22 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#23 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#24 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#25 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#26 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#27 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#28 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(87): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#29 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDuringMaintenance->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#30 /volume1/web/firefly-iii/app/Http/Middleware/SecureHeaders.php(51): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#31 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): FireflyIII\\Http\\Middleware\\SecureHeaders->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#32 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#33 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(141): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
#34 /volume1/web/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(110): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter(Object(Illuminate\\Http\\Request))
#35 /volume1/web/firefly-iii/public/index.php(76): Illuminate\\Foundation\\Http\\Kernel->handle(Object(Illuminate\\Http\\Request))
#36 {main}
"} 
[2020-12-30 23:03:45] local.INFO: Could not create or generate installation ID. Do not continue.  
```
- I was able to replicate the issue on the demo site.

**Comment by JC5** [comment]
Since you say "on the first load", does it work for consecutive loads or just never?

**Comment by Hauptstadt** [comment]
Sorry, I wasn't clear.  It never loads, it always gives me that error, but it is my initial attempt at loading /public/.

**Comment by JC5** [comment]
Mmm, that seems to indicate a missing mysql driver for PHP. Can you create a file called `pi.php` in the public directory and add the following content?

```php
<?php
phpinfo();
```

This file should have some info on the loaded (my)sql drivers.

**Comment by Hauptstadt** [comment]
Here's the snippet from the relevant MySQL sections and the modules below it
![image](https://user-images.githubusercontent.com/64873401/103412279-9d2a0b80-4b74-11eb-996b-409d2ff4d2bd.png)

![image](https://user-images.githubusercontent.com/64873401/103412313-c77bc900-4b74-11eb-9bea-90aa22a51c96.png)

**Comment by JC5** [comment]
There is no pdo_mysql? That's the one you need.

**Comment by Hauptstadt** [comment]
You nailed it.    Seems that I needed the pdo_mysql and openssl PHP plugins in addition to those you mention on the self-hosted server instructions.   

Thanks so much for all your patience and help, after two months of waiting on Symantec to update their NAS Software and struggling through my own ineptitude, I've gotten to add a bank account!  I'm excited to see what is ahead!
